### PR TITLE
Use SNAPSHOT for release when snapshot version

### LINKED ISSUE #971: Support -SNAPSHOT versions when generating RPMs by default
## Expected behaviour

Running `rpm:packageBin` with a version of 1.0.0-SNAPSHOT produces an rpm with name myapp-1.0.0-SNAPSHOT.rpm

## Actual behaviour

Running `rpm:packageBin` with a version of 1.0.0-SNAPSHOT produces a stack trace as hyphens are not supported in RPM versions

## Information

Using version 1.1.5 with SBT 0.13.5.2 on Fedora.

It is possible to work round this by manually setting `version in Rpm` and `rpmRelease`, however it would be nice if the default behaviour was to do this automatically, as the Maven RPM plugin does.

### PR DESCRIPTION
**Pull request by keirlawson**
Fixes #971 - feedback appreciated!

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @keirlawson,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by keirlawson** [comment]
I have signed the Lightbend CLA

File: src/sbt-test/rpm/snapshot-rpm/build.sbt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+packageDescription := """A fun package description of our software,
+  with multiple lines."""
+
+rpmVendor := "typesafe"
+
+rpmUrl := Some("http://github.com/sbt/sbt-native-packager")
+
+rpmLicense := Some("BSD")
+
+
```
Comment (muuki88) [inline src/sbt-test/rpm/snapshot-rpm/build.sbt]: A simple test task checking the `rpmVersion` would be nice. E.g. something like

```scala
TaskKey[Unit]("unzip") := {
  assert(rpmRelease.value === "SNAPSHOT", s"RPM has incorrect value ${rpmVersion.value}")
  assert(rpmMetadata.value.version === "...", s"RPM has incorrect value ${rpmMetadata.value.version}")
  ()
}
```

File: src/main/scala/com/typesafe/sbt/packager/rpm/RpmPlugin.scala [hunk 2]
```diff
@@ -59,11 +59,16 @@ object RpmPlugin extends AutoPlugin {
 
   }
 
+  private def stripSnapshot(version: String): String = {
```
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/rpm/RpmPlugin.scala]: I would put the private methods at the bottom of the AutoPlugin to preserve a "general -> detailed" structure :)

Comment (keirlawson) [inline src/main/scala/com/typesafe/sbt/packager/rpm/RpmPlugin.scala]: Done

**Comment by keirlawson** [comment]
Have moved helper method and added tests as suggested.

File: src/main/scala/com/typesafe/sbt/packager/rpm/RpmPlugin.scala (hunk 2 above)
Comment (keirlawson) [inline src/main/scala/com/typesafe/sbt/packager/rpm/RpmPlugin.scala]: Done

File: src/sbt-test/rpm/snapshot-rpm/build.sbt (hunk 1 above)
Comment (keirlawson) [inline src/sbt-test/rpm/snapshot-rpm/build.sbt]: Done

**Pull request merged**